src/js/App/analytics/segment-analytics.tsx

```diff
@@ -0,0 +1,122 @@
+import React, { createContext, useContext, useEffect, useRef } from 'react';
+import { AnalyticsBrowser } from '@segment/analytics-next';
+import { isProd } from '../../utils';
+import { useSelector } from 'react-redux';
+import { AnyObject } from '../../types';
+
+type SegmentEnvs = 'dev' | 'prod';
+type SegmentModules = 'openshift';
+
+const KEY_FALLBACK = {
+  prod: 'nm7VsnYsBVJ9MqjaVInft69pAkhCXq9Q',
+  dev: 'Aoak9IFNixtkZJRatfZG9cY1RHxbATW1',
+};
+
+function getAdobeVisitorId() {
+  const visitor = window?.s?.visitor;
+  if (visitor) {
+    return visitor.getMarketingCloudVisitorID();
+  }
+
+  return -1;
+}
+
+const getAPIKey = (env: SegmentEnvs = 'dev', module: SegmentModules) =>
+  ({
+    prod: {
+      openshift: 'z3Ic4EtzJtHrhXfpKgViJmf2QurSxXb9',
+    },
+    dev: {
+      openshift: 'A8iCO9n9Ax9ObvHBgz4hMC9htKB0AdKj',
+    },
+  }[env]?.[module] || KEY_FALLBACK[env]);
+
+const registerUrlObserver = () => {
+  /**
+   * We ignore hash changes
+   * Hashes only have frontend effect
+   */
+  let oldHref = document.location.href.replace(/#.*$/, '');
+
+  const bodyList = document.body;
+  const observer = new MutationObserver((mutations) => {
+    mutations.forEach(() => {
+      const newLocation = document.location.href.replace(/#.*$/, '');
+      if (oldHref !== newLocation) {
+        oldHref = newLocation;
+        window?.sendCustomEvent?.('pageBottom');
+        setTimeout(() => {
+          window.segment?.page(window?._segment?.pageOptions);
+        });
+      }
+    });
+  });
+  observer.observe(bodyList, {
+    childList: true,
+    subtree: true,
+  });
+  return observer.disconnect;
+};
+
+export const SegmentContext = createContext<{ ready: boolean; analytics?: AnalyticsBrowser }>({
+  ready: false,
+  analytics: undefined,
+});
+
+export type SegmentProviderProps = {
+  activeModule: string;
+};
+
+export const SegmentProvider: React.FC<SegmentProviderProps> = ({ activeModule, children }) => {
+  const analytics = useRef<AnalyticsBrowser>();
+  const user = useSelector(({ chrome: { user } }: AnyObject) => user);
+  useEffect(() => {
+    const disconnect = registerUrlObserver();
+    return () => disconnect();
+  }, []);
+
+  useEffect(() => {
+    if (activeModule && user) {
+      const newKey = getAPIKey(isProd() ? 'prod' : 'dev', activeModule as SegmentModules);
+      const identityOptions = { cloud_user_id: user.identity.internal.account_id, adobe_cloud_visitor_id: getAdobeVisitorId() };
+      const groupOptions = {
```

Comment:
Here's a fairly typical output from the current group call:

```
"account": {
  "id": "5583404",
  "account_number": "5583404",
  "account_id": "7378958",
  "cloud_org_id": "7378958",
  "cloud_ebs_id": "5583404"
}
```

Again, unfortunately trying to preserve that <sigh>, but also switching to `org_id` as the primary key for Group ID:

```
const groupTraits = {
  account_number: user.identity.account_number,
  account_id: user.identity.internal.org_id,
  cloud_org_id: user.identity.internal.org_id,
  cloud_ebs_id: user.identity.account_number
};
//...
analytics.current.group(
  user.identity.internal.org_id // switching to org_id as the Group ID
  groupTraits
);
```